### ISSUE TITLE
Error watching files with pattern "**/*.js"

### ISSUE THREAD

**Issue by bryzaguy** [paulrayes/gaze-cli#1]
I'd like to watch all js files in my current directory and any subdirectories but I get the following error when attempting to run gaze with the pattern '_**/_*.js':

``` bash
/usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/gaze.js:331
      return this._handleError(err);
                  ^
RangeError: Maximum call stack size exceeded
    at Gaze._pollFile (/usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/gaze.js:331:19)
    at /usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/gaze.js:411:12
    at Array.forEach (native)
    at /usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/gaze.js:409:11
    at iterate (/usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/helper.js:52:5)
    at /usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/helper.js:61:11
    at /usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/gaze.js:420:5
    at iterate (/usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/helper.js:52:5)
    at /usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/helper.js:61:11
    at /usr/local/lib/node_modules/gaze-cli/node_modules/gaze/lib/gaze.js:420:5
```

**Comment by paulrayes** [comment]
It looks like it's using polling instead of the OS built-in file watching, which doesn't work very well in Gaze. This happens when the number of files you are trying to watch is greater than your OS limit.

What OS are you using? OS X has a very low default limit of open file descriptors. If it's low you should increase it, in Ubuntu that would be `ulimit -n 65535`, I'm not sure about in OS X. To check what it is use just `ulimit -n`.

It's also generally not a good idea to watch all Javascript files, you should just watch the files you are using. This both makes it perform better and decreases the chance of you reaching that limit. If your code is in a lib folder you can change your pattern to `lib/**/*.js`, or you can exclude node_modules with `!node_modules/**/*.js`.

Let me know if either of those fixes the problem. If neither fixes it, please let me know your OS, Node version, and approximately how many files you're trying to watch.

**Comment by paulrayes** [comment]
This may be helpful: http://stackoverflow.com/questions/5377450/maximum-number-of-open-filehandles-per-process-on-osx-and-how-to-increase

**Comment by bryzaguy** [comment]
I reduced the number of files in node_modules and it worked.  Is there a way to provide multiple patterns so I can do !node_modules/**/*.js and **/*.js?

**Comment by bryzaguy** [comment]
Also FYI I had like 2500 files in my open files setting on OS X.

**Comment by paulrayes** [comment]
With a single pattern that's negated, it will watch everything that does not match the pattern, but pull request #2 adds support for multiple patterns which will be in the next version.

I'll add a note to the readme about this limit.

**Comment by paulrayes** [comment]
Released v0.2.0 with a note about this file limit, and support for multiple patterns
